internal/testtypes/builders.go

```diff
@@ -0,0 +1,525 @@
+	obj := &clusterv1.ClusterClass{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ClusterClass",
+			APIVersion: clusterv1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      c.name,
+			Namespace: c.namespace,
+		},
+		Spec: clusterv1.ClusterClassSpec{},
+	}
+	if c.infrastructureClusterTemplate != nil {
+		obj.Spec.Infrastructure = clusterv1.LocalObjectTemplate{
+			Ref: objToRef(c.infrastructureClusterTemplate),
+		}
+	}
+	if c.controlPlaneMetadata != nil {
+		obj.Spec.ControlPlane.Metadata = *c.controlPlaneMetadata
+	}
+	if c.controlPlaneTemplate != nil {
+		obj.Spec.ControlPlane.LocalObjectTemplate = clusterv1.LocalObjectTemplate{
+			Ref: objToRef(c.controlPlaneTemplate),
+		}
+	}
+	if c.controlPlaneInfrastructureMachineTemplate != nil {
+		obj.Spec.ControlPlane.MachineInfrastructure = &clusterv1.LocalObjectTemplate{
+			Ref: objToRef(c.controlPlaneInfrastructureMachineTemplate),
+		}
+	}
+	if len(c.machineDeploymentClasses) > 0 {
+		obj.Spec.Workers.MachineDeployments = append(obj.Spec.Workers.MachineDeployments, c.machineDeploymentClasses...)
+	}
+	return obj
+}
+
+// MachineDeploymentClassBuilder holds the variables and objects required to build a clusterv1.MachineDeploymentClass.
+type MachineDeploymentClassBuilder struct {
+	namespace                     string
+	name                          string
+	replicas                      int32
+	class                         string
+	infrastructureMachineTemplate *unstructured.Unstructured
+	bootstrapTemplate             *unstructured.Unstructured
+	labels                        map[string]string
+	annotations                   map[string]string
+}
+
+// NewMachineDeploymentClassBuilder returns a MachineDeploymentClassBuilder with the given name and namespace.
+func NewMachineDeploymentClassBuilder(namespace, name string) *MachineDeploymentClassBuilder {
+	return &MachineDeploymentClassBuilder{
+		name:      name,
+		namespace: namespace,
+	}
+}
+
+// WithInfrastructureTemplate registers the passed Unstructured object as the InfrastructureMachineTemplate for the MachineDeploymentClassBuilder.
+func (m *MachineDeploymentClassBuilder) WithInfrastructureTemplate(t *unstructured.Unstructured) *MachineDeploymentClassBuilder {
+	m.infrastructureMachineTemplate = t
+	return m
+}
+
+// WithBootstrapTemplate registers the passed Unstructured object as the BootstrapTemplate for the MachineDeploymentClassBuilder.
+func (m *MachineDeploymentClassBuilder) WithBootstrapTemplate(t *unstructured.Unstructured) *MachineDeploymentClassBuilder {
+	m.bootstrapTemplate = t
+	return m
+}
+
+// WithClass sets the MachineDeploymentClass name for the MachineDeploymentClassBuilder.
+func (m *MachineDeploymentClassBuilder) WithClass(class string) *MachineDeploymentClassBuilder {
+	m.class = class
+	return m
+}
+
+// WithReplicas sets the number of replicas for the MachineDeploymentClassBuilder.
+func (m *MachineDeploymentClassBuilder) WithReplicas(replicas int32) *MachineDeploymentClassBuilder {
```

Comment:
replicas are not used when building the MachineDeploymentClass. If I'm correct it, we can just drop `WithReplicas` because replicas are not part of the MachineDeploymentClass